util/file.py

```diff
@@ -128,6 +128,16 @@ def tmp_dir(*args, **kwargs):
         else:
             shutil.rmtree(name)
 
+@contextlib.contextmanager
+def tmp_chdir(name):
```

Comment:
This isn't `tmp`-specific, right? (I get what you mean: temporarily change directories within the generator context, but it reads like it deals with temp files) Let's rename it to be more generic. How about `pushd()`, `pushd_and_popd()` or something along those lines? Let's also change the argument `name` to `path`. Perhaps it should `yield path` for convenience downstream.